controller/konnect/ops.go

```diff
@@ -111,3 +131,27 @@ func logOpComplete[
 		"konnect_id", e.GetKonnectStatus().GetKonnectID(),
 	)
 }
+
+// handleResp checks the response from the Konnect API and returns an error if
```

Comment:
I have a question regarding the SDK. Methods like `CreateControlPlane` always return a response and an error. Why do we need to handle the raw response body again on our own? Shouldn't the SDK do it for us and return a properly parsed error already?

Reply:
We could leverage what the sdk offers us. `handleResp` is somewhat of a syntactic sugar to generically use it for all error handling. Do you believe this could be replaced with something more readable, more succinct?

I'm curious what we can improve here.

Reply:
We could probably skip `handleResp` as the SDK already reads the body and maps it to specific error types. That means that there should be no case where an SDK method returns `err == nil` and a non-successful `resp.GetStatusCode()`, making the `handleResp` redundant and noop (as it will always return on the first `if err != nil`). Please correct me if I'm wrong.

Reply:
Agreed. I've changed `handleResp` to basically be only a uniform error wrapper. HTTP body was also already read at that point ( SDK functions read the body to get the error details). 

PTAL 02f5c354dde7c6fa39e5484a0cbcb01358726859